community/consistency-check/src/main/java/org/neo4j/consistency/statistics/AccessStatistics.java

```diff
@@ -0,0 +1,179 @@
+ */
+package org.neo4j.consistency.statistics;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.neo4j.kernel.impl.store.RecordStore;
+import org.neo4j.kernel.impl.store.record.AbstractBaseRecord;
+
+import static java.lang.String.format;
+
+/**
+ * Keeps access statistics about a store, i.e. identifying {@link RecordStore#getRecord(long)} patterns
+ * and how random the access is.
+ */
+public class AccessStatistics
+{
+    public static int LOCALITY = 500;
+
+    private final Map<RecordStore<? extends AbstractBaseRecord>,AccessStats> stats = new HashMap<>();
+    private int proximity;
+
+    @SuppressWarnings( "unchecked" )
+    public AccessStats getAccessStats( RecordStore<? extends AbstractBaseRecord> store )
+    {
+        return stats.get( store );
+    }
+
+    public void register( RecordStore<? extends AbstractBaseRecord> store, AccessStats accessStats )
+    {
+        assert !stats.containsKey( store );
+        stats.put( store, accessStats );
+    }
+
+    public String getAccessStatSummary()
+    {
+        String msg = "";
+        for ( AccessStats accessStats : stats.values() )
+        {
+            String accessStat = accessStats.toString();
+            if ( accessStat.length() != 0 )
+            {
+                msg += format( accessStat + "%n" );
+            }
+        }
+        return msg;
+    }
+
+    public void reset()
+    {
+        for ( AccessStats accessStats : stats.values() )
+        {
+            accessStats.reset();
+        }
+    }
+
+    public static class AccessStats
+    {
+        private long reads = 0, writes = 0, inUse = 0;
+        private long randomReads = 0, randomWrites = 0;
+        private int proximityValue = 0;
+        private final String storeType;
+        private long prevReadId, prevWriteId;
+
+        public AccessStats( String type, int proximity )
+        {
+            this.storeType = type;
+            this.proximityValue = proximity;
+        }
+
+        @Override
+        public String toString()
+        {
+            if ( reads == 0 && writes == 0 && randomReads == 0 )
+            {
+                return "";
+            }
+            StringBuilder buf = new StringBuilder( storeType );
+            appendStat( buf, "InUse", inUse );
+            appendStat( buf, "Reads", reads );
+            appendStat( buf, "Random Reads", randomReads );
+            appendStat( buf, "Writes", writes );
+            appendStat( buf, "Random Writes", randomWrites );
+            int scatterIndex = 0;
+            if ( randomReads > 0 )
+            {
+                long scatterReads = reads == 0 ? randomReads : reads;
+                scatterIndex = (int) ((randomReads * 100) / scatterReads);
+            }
+            appendStat( buf, "ScatterIndex", scatterIndex );
+            if ( scatterIndex > 0.5 )
+            {
+                buf.append( format( "%n *** Property Store reorgization is recommended for optimal performance ***" ) );
```

Comment:
probably we should disable this suggestion for now till we will be able to do so